lib/course_planner_web/controllers/auth/api_session_controller.ex

```diff
@@ -0,0 +1,41 @@
+defmodule CoursePlannerWeb.Auth.Api.V1.JsonSessionController do
+  @moduledoc """
+    This module handles api loging in to the system
+  """
+  use CoursePlannerWeb, :controller
+
+  alias CoursePlanner.Accounts.{Users, User}
+  alias Guardian.Plug
+
+  plug :put_layout, ""
+
+  def create(conn, %{"email" => email, "password" => password}) do
+    trimmed_downcased_email =
+      email
+      |> String.trim()
+      |> String.downcase()
+    user = Repo.get_by(User, email: trimmed_downcased_email)
+
+    case Users.check_password(user, password) do
+      {:ok, _reason} ->
+        conn
+        |> json(%{token: get_login_token(conn, user)})
+
+      {:error, _reason} ->
+        conn
+        |> json(%{token:  "error"})
+    end
+  end
+  def create(conn, _params) do
+    conn
+    |> put_status(406)
+    |> render(CoursePlannerWeb.ErrorView, "406.json", errors: [])
+  end
+
+  defp get_login_token(conn, user) do
+    conn
+    |> Plug.api_sign_in(user)
```

Comment:
Since now there's gonna be a public API for login, maybe we need the ability to revoke tokens.
Thoughts on that? Will need to set something like `GuardianDb`.

Reply:
Agreed.
As I have seen this is a functionality provided by guardianDB.
So what do you suggest ? to have it under this scope or another PR ?

Reply:
Another PR. We should see if we'll need the revoke functionality.

Reply:
Card added
https://app.nostromo.io/work/course-planner/investigate-and-add-revoke-functionality-for-login-tokens-is-decided?e21vZHVsZTp3b3JrLHByb2plY3Q6NjM2OCxjYXJkOjk4MDIzLGNhcmRfdGl0bGU6aW52ZXN0aWdhdGUtYW5kLWFkZC1yZXZva2UtZnVuY3Rpb25hbGl0eS1mb3ItbG9naW4tdG9rZW5zLWlzLWRlY2lkZWR9